source/validate_cfg.cpp

```diff
@@ -424,14 +424,63 @@ spv_result_t PerformCfgChecks(ValidationState_t& _) {
       for (auto edge : postdom_edges) {
         edge.first->SetImmediatePostDominator(edge.second);
       }
-      /// calculate back edges.
+      /// Calculate back edges.
       DepthFirstTraversal(
           function.pseudo_entry_block(),
           function
               .AugmentedCFGSuccessorsFunctionIncludingHeaderToContinueEdge(),
           ignore_block, ignore_block, [&](cbb_ptr from, cbb_ptr to) {
             back_edges.emplace_back(from->id(), to->id());
           });
+
+      // Perform a DepthFirstTraversal that ensures CFG nesting depth is valid.
```

Comment:
You should explain in words how you determine depth of a block:
- It's 0 for the function entry block
- the non-header blocks in a loop or selection construct (that aren't in yet a deeper nested construct) have 1 more than the header
- the blocks in a continue construct (that aren't in a deeper nested construct) have 1 more than the header